flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/recovery/DemultiplexingRecordDeserializerTest.java

```diff
@@ -123,7 +122,8 @@ public void testUpscale() throws IOException {
 
             assertEquals(
                     Arrays.asList(start + 1L, start + 2L, start + 3L), readLongs(deserializer));
-            assertTrue(bufferBuilder.getMemorySegment().isFreed());
+
+            bufferBuilder.recycle();
```

Comment:
What is this revert about? I can't connect it to any other change in this PR.

Reply:
Oh, shoot. I reverted a wrong commit :facepalm:  It was supposed to be the one that ignores the test. I'll have to rerun Azure...

Reply:
I was too quick and too careless again :(